hack/generated/go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/Azure/azure-service-operator/hack/generated
 go 1.17
 
 require (
+	github.com/Azure/azure-service-operator/v2 v2.0.0-00010101000000-000000000000
```

Comment:
I feel like this is actually a dependency where it's going to GitHub to download it? Rather than getting it locally?

Reply:
I know you didn't break this here, just highlighting